R/metrics.R

```diff
@@ -568,3 +574,13 @@ eval_safely <- function(expr, expr_nm, data = NULL, env = caller_env()) {
     }
   )
 }
+
+
+call_remove_duplicate_arguments <- function(call) {
+  arg_names <- environment(call[[1]])[["fixed"]]
```

Comment:
This is mainly what worries me.

What if `metric_tweak()` stored the names of the "static" tweaked arguments as an attribute of the function it returned? i.e. imagine that `metric_tweak("f_meas_weighted", f_meas, estimator = "macro_weighted")` stored `"estimator"` as an attribute named `static` on the result. Then you could do:

```r
call_remove_static_arguments <- function(call, fn) {
  static <- get_static_arguments(fn)
  
  if (length(static) == 0L) {
    # No static arguments
    return(call)
  }
  
  names <- rlang::call_args_names(call)
  names <- intersect(names, static)
  
  if (length(names) == 0L) {
    # `static` arguments don't intersect with `call`
    return(call)
  }
  
  zaps <- rlang::rep_named(names, list(rlang::zap()))
  call <- call_modify(call, !!!zaps)
  
  call
}
```

And in the file where `metric_tweak()` lives you could define these two helpers to help you get and set the `static` attribute

```r
get_static_arguments <- function(fn) {
  attr(fn, "static", exact = TRUE)
}
set_static_arguments <- function(fn, static) {
  attr(fn, "static") <- static
  fn
}
```